modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectLayoutResourceImpl.java

```diff
@@ -306,7 +324,7 @@ private com.liferay.object.model.ObjectLayoutTab _toObjectLayoutTab(
 				objectLayoutBox -> _toObjectLayoutBox(
 					objectDefinitionId, objectLayoutBox)));
 		serviceBuilderObjectLayoutTab.setObjectRelationshipId(
-			GetterUtil.getLong(objectLayoutTab.getObjectRelationshipId()));
+			objectRelationshipId);
```

Comment:
As used, i.e., declare `objectRelationshipId` as close as possible to its first line of usage:

```
long objectRelationshipId = GetterUtil.getLong(
	objectLayoutTab.getObjectRelationshipId());

if (Validator.isNotNull(
	objectLayoutTab.getObjectRelationshipExternalReferenceCode())) {

	ObjectRelationship objectRelationship =
		_objectRelationshipLocalService.
			getObjectRelationshipByExternalReferenceCode(
				objectLayoutTab.
					getObjectRelationshipExternalReferenceCode(),
				contextCompany.getCompanyId(), objectDefinitionId);

	objectRelationshipId = objectRelationship.getObjectRelationshipId();
}

serviceBuilderObjectLayoutTab.setObjectRelationshipId(
	objectRelationshipId);
```

OR

```
if (Validator.isNotNull(
		objectLayoutTab.getObjectRelationshipExternalReferenceCode())) {

	ObjectRelationship objectRelationship =
		_objectRelationshipLocalService.
			getObjectRelationshipByExternalReferenceCode(
				objectLayoutTab.
					getObjectRelationshipExternalReferenceCode(),
				contextCompany.getCompanyId(), objectDefinitionId);

	serviceBuilderObjectLayoutTab.setObjectRelationshipId(
		objectRelationship.getObjectRelationshipId());
}
else {
	serviceBuilderObjectLayoutTab.setObjectRelationshipId(
		GetterUtil.getLong(objectLayoutTab.getObjectRelationshipId()));
}
```